src/models/connectionStore.ts

```diff
@@ -158,11 +158,9 @@ export class ConnectionStore {
      * @param connectionCredentials Connection credentials of profile for password lookup
      */
     public async lookupPassword(connectionCredentials: IConnectionCredentials): Promise<string> {
-        const databaseName = connectionCredentials.database === '' ? 'master' :
-            connectionCredentials.database;
         const credentialId = ConnectionStore.formatCredentialId(
-            connectionCredentials.server, databaseName,
-            connectionCredentials.user, ConnectionStore.CRED_MRU_USER);
+            connectionCredentials.server, connectionCredentials.database,
+            connectionCredentials.user, ConnectionStore.CRED_PROFILE_USER);
```

Comment:
>CRED_PROFILE_USER [](start = 56, length = 17)

So it looks like we're saving the password in either the MRU or PROFILE stores - so shouldn't the lookup be looking in both possibly? Although I guess I'd wonder why we'd even need to store the password from the MRU separately - aren't all connections added to the profile anyways?

Reply:
I'm not sure what the design was when this was written, but the MRU saved every single connection made to it and didn't respect `savePassword`. I fixed that bug now, and get the password from User Profile, if it were saved.